src/wrappers.jl

```diff
@@ -0,0 +1,402 @@
+end
+
+macro apicall(name, args...)
+    quote
+        ptr = lookup_function($(name))
+        ptr == C_NULL && throw(MissingCollectorError())
+        ccall(ptr, ($(map(esc, args)...)))
+    end
+end
+
+@enum __itt_collection_state::Cint begin
+    __itt_collection_uninitialized = 0          # uninitialized
+    __itt_collection_init_fail = 1              # failed to init
+    __itt_collection_collector_absent = 2       # non work state collector is absent
+    __itt_collection_collector_exists = 3       # work state collector exists
+    __itt_collection_init_successful = 4        # success to init
+end
+
+collection_state() =
+    ccall((:__itt_get_collection_state, ittapi_jll.libittnotify), __itt_collection_state, ())
+
+isactive() = collection_state() == __itt_collection_init_successful
+
+
+#
+# Domains
+#
+
+export Domain, name, isenabled, enable!
+
+struct __itt_domain
+    flags::Cint
+    name::Cstring
+    # don't care about the rest, for now
+end
+
+struct Domain
+    handle::Ptr{__itt_domain}
+end
+Base.unsafe_convert(::Type{Ptr{__itt_domain}}, d::Domain) = d.handle
+
+Domain(name::String) = Domain(@apicall(:__itt_domain_create, Ptr{__itt_domain}, (Cstring,), name))
+
+# XXX: can we do this cleaner?
+function Base.getproperty(d::Domain, name::Symbol)
+    if name in [:flags, :name]
+        d.handle == C_NULL && throw(MissingCollectorError())
+        return getfield(unsafe_load(d.handle), name)
+    else
+        return getfield(d, name)
+    end
+end
+function Base.setproperty!(d::Domain, name::Symbol, value)
+    if name in [:flags]
+        d.handle == C_NULL && throw(UndefRefError())
+        idx = Base.fieldindex(__itt_domain, name)
+        offset = Base.fieldoffset(__itt_domain, idx)
+        typ = Base.fieldtype(__itt_domain, idx)
+        return unsafe_store!(convert(Ptr{Cint}, d.handle) + offset, value)
+    else
+        return setfield!(d, name, value)
+    end
+end
+
+function Base.show(io::IO, d::Domain)
+    print(io, "Domain(")
+    if isactive()
+        print(io, repr(name(d)), ", enabled=$(isenabled(d))")
+    end
+    print(io, ")")
+end
+
+isenabled(d::Domain) = d.flags == 0 ? false : true
+enable!(d::Domain, enable::Bool=true) = d.flags = enable ? 1 : 0
+
+name(d::Domain) = d.name == C_NULL ? "" : unsafe_string(d.name)
+
+
+#
+# String handles
+#
+
+export StringHandle
+
+struct __itt_string_handle
+    str::Cstring
+end
+
+struct StringHandle
+    handle::Ptr{__itt_string_handle}
+end
+Base.unsafe_convert(::Type{Ptr{__itt_string_handle}}, s::StringHandle) = s.handle
+
+StringHandle(name::String) =
+    StringHandle(@apicall(:__itt_string_handle_create, Ptr{__itt_string_handle}, (Cstring,), name))
+
+String(s::StringHandle) = s.handle == C_NULL ? "" : unsafe_string(unsafe_load(s.handle).str)
+
+
+#
+# Collection control
+#
+
+export pause, resume, detach
+
+pause() = @apicall(:__itt_pause, Cvoid, ())
+resume() = @apicall(:__itt_resume, Cvoid, ())
+detach() = @apicall(:__itt_detach, Cvoid, ())
```

Comment:
I left this intentionally unexposed since this actively breaks JITProfiline

Reply:
So working as intended?

> Detach data collection. VTune Profiler detaches all collectors from all processes. Your application continues to work but no data is collected for the running collection.